### DISCUSSION TITLE
Stats & graphs

### DISCUSSION THREAD

**Discussion by raphodn** [openfoodfacts/open-prices#334] [Ideas]
## Ideas

|model|detail|example|
|---|---|---|
|Product|Number of prices<br>Number of locations/countries<br>Map of locations (with prices)||
|Product|Evolution of price||
|Category|min, max, median price||
|Product|Evolution of price per location/store chain||
|Location|Number of prices<br>Number of different products<br>Number of different users||
|Location : city|Number of cities with at least 1 price<br>Cities with the most locations<br>Cities with the most prices||
|Location : country|Number of countries with at least 1 price<br>Countries with the most locations<br>Countries with the most prices||
|Location : type|Number of types with at least 1 price<br>Number of prices per type|`|
|Location : name|Number of prices per store name/chain||
|User|Number of prices<br>Number of different products<br>Number of different locations||
|Proof|Number of prices per type||

filters on date, location store, location country, user, category...

## Related issues

- #254 

## Related discussions

- #399

**Comment by serpico** [comment]
Related to #302 @raphodn could you pull some stats ( similar to what you did about brands to advocate in favor of a taxonomy, I think it was a few weeks ago ) of the number of prices per "supermarket chain" ? i.e Carrefour City, Carrefour Contact, U Express, Proxi, Spar, Franprix, etc...to see if there's a way to optionally filter based on the size of the shop or something to that effect, prices from those small supermarkets create spikes on the graphs.

edit : was it what you meant by **Location : type** in the table up top ? I saw Location : type on the wiki.

**Reply by raphodn** [reply]
Here you go, I did a basic aggregation of price location's "osm_name"
the top 30 results are here : https://github.com/openfoodfacts/open-prices/wiki/Stats#top-location-names

**Reply by raphodn** [reply]
the shop names (at least their homogenization) happens mostly here : https://github.com/osmlab/name-suggestion-index

that's why i opened a thread in the OSM forum a couple of weeks ago ^^
https://forum.openstreetmap.fr/t/aidez-moi-a-lister-rajouter-des-chaines-de-magasins-manquants-dans-nsi/26727

**Reply by raphodn** [reply]
location type : "shop:supermarket"
location name : "Carrefour City" (added in the table :+1: )

**Reply by raphodn** [reply]
makes me think of how to make these stats easily accessible/queryable :thinking: 
(without having to run SQL or django shell commands)
dedicated endpoints ?

**Comment by serpico** [comment]
Packaging size : ex: using both https://prices.openfoodfacts.org/app/categories/en:cocoa-and-hazelnuts-spreads and https://prices.openfoodfacts.org/app/brands/Nutella to show price/kg evolution depending on packaging size another rather striking difference ( I wasn't expecting ) between `3760335580666` and `3760197946945`. 

Maybe adding brand to the list of filters under the table would be enough ?

**Comment by serpico** [comment]
Filtering through quantity : if I'm looking for Tea in bulk inside https://prices.openfoodfacts.org/app/categories/en:green-teas I have a hunch anything < 100 g won't return what I'm looking for .

**Comment by serpico** [comment]
Ability to set a reference point date in user's preference ( pre-covid, pre-cocoa bean price surge, etc... ) and display a colored vertical bar on the graphs.

The price evolution ( primarily in %, maybe in currency ) would be displayed in the current tooltip  ( when the mouse pointer is over a price point in the graph view ) compared to that reference point.
